diff -urN a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -0,0 +1,17 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "glog.go",
+        "glog_file.go",
+    ],
+    importpath = "github.com/golang/glog",
+    visibility = ["//visibility:public"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["glog_test.go"],
+    embed = [":go_default_library"],
+)
