diff -urN d/google/api/BUILD.bazel e/google/api/BUILD.bazel
--- d/google/api/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ e/google/api/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,8 +12,13 @@
 
 proto_library(
     name = "api_proto",
-    srcs = ["launch_stage.proto"],
+    srcs = [
+        "experimental/authorization_config.proto",
+        "experimental/experimental.proto",
+        "launch_stage.proto",
+    ],
     visibility = ["//visibility:public"],
+    deps = [":annotations_proto"],
 )
 
 proto_library(
@@ -112,6 +117,7 @@
     importpath = "google.golang.org/genproto/googleapis/api",
     proto = ":api_proto",
     visibility = ["//visibility:public"],
+    deps = [":annotations_go_proto"],
 )
 
 go_proto_library(
diff -urN d/google/api/experimental/BUILD.bazel e/google/api/experimental/BUILD.bazel
--- d/google/api/experimental/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ e/google/api/experimental/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
@@ -1,25 +0,0 @@
-load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
-
-proto_library(
-    name = "api_proto",
-    srcs = [
-        "authorization_config.proto",
-        "experimental.proto",
-    ],
-    visibility = ["//visibility:public"],
-    deps = [
-        "//google/api:annotations_proto",
-        "//google/api:api_proto",
-    ],
-)
-
-go_proto_library(
-    name = "api_go_proto",
-    importpath = "google.golang.org/genproto/googleapis/api",
-    proto = ":api_proto",
-    visibility = ["//visibility:public"],
-    deps = [
-        "//google/api:annotations_go_proto",
-        "//google/api:api_go_proto",
-    ],
-)
