diff -urN b/internal/channelz/BUILD.bazel c/internal/channelz/BUILD.bazel
--- b/internal/channelz/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/internal/channelz/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -17,6 +17,15 @@
         "//credentials:go_default_library",
         "//grpclog:go_default_library",
     ] + select({
+        "@io_bazel_rules_go//go/platform:darwin": [
+            "@org_golang_x_sys//unix:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:windows": [
+            "@org_golang_x_sys//unix:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:android": [
+            "@org_golang_x_sys//unix:go_default_library",
+        ],
         "@io_bazel_rules_go//go/platform:linux": [
             "@org_golang_x_sys//unix:go_default_library",
         ],
@@ -29,6 +38,9 @@
     srcs = ["util_test.go"],
     embed = [":go_default_library"],
     deps = select({
+        "@io_bazel_rules_go//go/platform:android": [
+            "@org_golang_x_sys//unix:go_default_library",
+        ],
         "@io_bazel_rules_go//go/platform:linux": [
             "@org_golang_x_sys//unix:go_default_library",
         ],
