diff -urN c/go/analysis/BUILD.bazel d/go/analysis/BUILD.bazel
--- c/go/analysis/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -11,3 +11,9 @@
     importpath = "golang.org/x/tools/go/analysis",
     visibility = ["//visibility:public"],
 )
+
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
diff -urN c/go/analysis/internal/facts/BUILD.bazel d/go/analysis/internal/facts/BUILD.bazel
--- c/go/analysis/internal/facts/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/internal/facts/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -7,13 +7,22 @@
         "imports.go",
     ],
     importpath = "golang.org/x/tools/go/analysis/internal/facts",
-    visibility = ["//go/analysis:__subpackages__"],
+    visibility = [
+        "//go/analysis:__subpackages__",
+        "@io_bazel_rules_go//go/tools/builders:__pkg__",
+    ],
     deps = [
         "//go/analysis:go_default_library",
         "//go/types/objectpath:go_default_library",
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["facts_test.go"],
diff -urN c/go/analysis/passes/asmdecl/BUILD.bazel d/go/analysis/passes/asmdecl/BUILD.bazel
--- c/go/analysis/passes/asmdecl/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/asmdecl/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -11,6 +11,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["asmdecl_test.go"],
diff -urN c/go/analysis/passes/assign/BUILD.bazel d/go/analysis/passes/assign/BUILD.bazel
--- c/go/analysis/passes/assign/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/assign/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["assign_test.go"],
diff -urN c/go/analysis/passes/atomic/BUILD.bazel d/go/analysis/passes/atomic/BUILD.bazel
--- c/go/analysis/passes/atomic/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/atomic/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["atomic_test.go"],
diff -urN c/go/analysis/passes/atomicalign/BUILD.bazel d/go/analysis/passes/atomicalign/BUILD.bazel
--- c/go/analysis/passes/atomicalign/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/atomicalign/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["atomicalign_test.go"],
diff -urN c/go/analysis/passes/atomicalign/BUILD.bazel.orig d/go/analysis/passes/atomicalign/BUILD.bazel.orig
--- c/go/analysis/passes/atomicalign/BUILD.bazel.orig	1969-12-31 19:00:00.000000000 -0500
+++ d/go/analysis/passes/atomicalign/BUILD.bazel.orig	2000-01-01 00:00:00.000000000 -0000
@@ -0,0 +1,21 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["atomicalign.go"],
+    importpath = "golang.org/x/tools/go/analysis/passes/atomicalign",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//go/analysis:go_default_library",
+        "//go/analysis/passes/inspect:go_default_library",
+        "//go/analysis/passes/internal/analysisutil:go_default_library",
+        "//go/ast/inspector:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["atomicalign_test.go"],
+    embed = [":go_default_library"],
+    deps = ["//go/analysis/analysistest:go_default_library"],
+)
diff -urN c/go/analysis/passes/bools/BUILD.bazel d/go/analysis/passes/bools/BUILD.bazel
--- c/go/analysis/passes/bools/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/bools/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["bools_test.go"],
diff -urN c/go/analysis/passes/buildssa/BUILD.bazel d/go/analysis/passes/buildssa/BUILD.bazel
--- c/go/analysis/passes/buildssa/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/buildssa/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -11,6 +11,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["buildssa_test.go"],
diff -urN c/go/analysis/passes/buildtag/BUILD.bazel d/go/analysis/passes/buildtag/BUILD.bazel
--- c/go/analysis/passes/buildtag/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/buildtag/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -11,6 +11,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["buildtag_test.go"],
diff -urN c/go/analysis/passes/cgocall/BUILD.bazel d/go/analysis/passes/cgocall/BUILD.bazel
--- c/go/analysis/passes/cgocall/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/cgocall/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -11,6 +11,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["cgocall_test.go"],
diff -urN c/go/analysis/passes/composite/BUILD.bazel d/go/analysis/passes/composite/BUILD.bazel
--- c/go/analysis/passes/composite/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/composite/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -15,6 +15,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["composite_test.go"],
diff -urN c/go/analysis/passes/copylock/BUILD.bazel d/go/analysis/passes/copylock/BUILD.bazel
--- c/go/analysis/passes/copylock/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/copylock/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["copylock_test.go"],
diff -urN c/go/analysis/passes/ctrlflow/BUILD.bazel d/go/analysis/passes/ctrlflow/BUILD.bazel
--- c/go/analysis/passes/ctrlflow/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/ctrlflow/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -14,6 +14,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["ctrlflow_test.go"],
diff -urN c/go/analysis/passes/deepequalerrors/BUILD.bazel d/go/analysis/passes/deepequalerrors/BUILD.bazel
--- c/go/analysis/passes/deepequalerrors/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/deepequalerrors/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["deepequalerrors_test.go"],
diff -urN c/go/analysis/passes/errorsas/BUILD.bazel d/go/analysis/passes/errorsas/BUILD.bazel
--- c/go/analysis/passes/errorsas/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/errorsas/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["errorsas_test.go"],
diff -urN c/go/analysis/passes/findcall/BUILD.bazel d/go/analysis/passes/findcall/BUILD.bazel
--- c/go/analysis/passes/findcall/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/findcall/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -8,6 +8,12 @@
     deps = ["//go/analysis:go_default_library"],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["findcall_test.go"],
diff -urN c/go/analysis/passes/httpresponse/BUILD.bazel d/go/analysis/passes/httpresponse/BUILD.bazel
--- c/go/analysis/passes/httpresponse/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/httpresponse/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["httpresponse_test.go"],
diff -urN c/go/analysis/passes/httpresponse/BUILD.bazel.orig d/go/analysis/passes/httpresponse/BUILD.bazel.orig
--- c/go/analysis/passes/httpresponse/BUILD.bazel.orig	1969-12-31 19:00:00.000000000 -0500
+++ d/go/analysis/passes/httpresponse/BUILD.bazel.orig	2000-01-01 00:00:00.000000000 -0000
@@ -0,0 +1,21 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["httpresponse.go"],
+    importpath = "golang.org/x/tools/go/analysis/passes/httpresponse",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//go/analysis:go_default_library",
+        "//go/analysis/passes/inspect:go_default_library",
+        "//go/analysis/passes/internal/analysisutil:go_default_library",
+        "//go/ast/inspector:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["httpresponse_test.go"],
+    embed = [":go_default_library"],
+    deps = ["//go/analysis/analysistest:go_default_library"],
+)
diff -urN c/go/analysis/passes/inspect/BUILD.bazel d/go/analysis/passes/inspect/BUILD.bazel
--- c/go/analysis/passes/inspect/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/inspect/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -10,3 +10,9 @@
         "//go/ast/inspector:go_default_library",
     ],
 )
+
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
diff -urN c/go/analysis/passes/internal/analysisutil/BUILD.bazel d/go/analysis/passes/internal/analysisutil/BUILD.bazel
--- c/go/analysis/passes/internal/analysisutil/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/internal/analysisutil/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -6,3 +6,9 @@
     importpath = "golang.org/x/tools/go/analysis/passes/internal/analysisutil",
     visibility = ["//go/analysis/passes:__subpackages__"],
 )
+
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
diff -urN c/go/analysis/passes/loopclosure/BUILD.bazel d/go/analysis/passes/loopclosure/BUILD.bazel
--- c/go/analysis/passes/loopclosure/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/loopclosure/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,6 +12,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["loopclosure_test.go"],
diff -urN c/go/analysis/passes/lostcancel/BUILD.bazel d/go/analysis/passes/lostcancel/BUILD.bazel
--- c/go/analysis/passes/lostcancel/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/lostcancel/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -14,6 +14,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["lostcancel_test.go"],
diff -urN c/go/analysis/passes/nilfunc/BUILD.bazel d/go/analysis/passes/nilfunc/BUILD.bazel
--- c/go/analysis/passes/nilfunc/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/nilfunc/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,6 +12,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["nilfunc_test.go"],
diff -urN c/go/analysis/passes/nilness/BUILD.bazel d/go/analysis/passes/nilness/BUILD.bazel
--- c/go/analysis/passes/nilness/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/nilness/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,6 +12,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["nilness_test.go"],
diff -urN c/go/analysis/passes/pkgfact/BUILD.bazel d/go/analysis/passes/pkgfact/BUILD.bazel
--- c/go/analysis/passes/pkgfact/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/pkgfact/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -8,6 +8,12 @@
     deps = ["//go/analysis:go_default_library"],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["pkgfact_test.go"],
diff -urN c/go/analysis/passes/printf/BUILD.bazel d/go/analysis/passes/printf/BUILD.bazel
--- c/go/analysis/passes/printf/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/printf/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -17,6 +17,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["printf_test.go"],
diff -urN c/go/analysis/passes/shadow/BUILD.bazel d/go/analysis/passes/shadow/BUILD.bazel
--- c/go/analysis/passes/shadow/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/shadow/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,6 +12,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["shadow_test.go"],
diff -urN c/go/analysis/passes/shift/BUILD.bazel d/go/analysis/passes/shift/BUILD.bazel
--- c/go/analysis/passes/shift/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/shift/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -16,6 +16,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["shift_test.go"],
diff -urN c/go/analysis/passes/sortslice/BUILD.bazel d/go/analysis/passes/sortslice/BUILD.bazel
--- c/go/analysis/passes/sortslice/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/sortslice/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["analyzer_test.go"],
diff -urN c/go/analysis/passes/stdmethods/BUILD.bazel d/go/analysis/passes/stdmethods/BUILD.bazel
--- c/go/analysis/passes/stdmethods/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/stdmethods/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,6 +12,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["stdmethods_test.go"],
diff -urN c/go/analysis/passes/stringintconv/BUILD.bazel d/go/analysis/passes/stringintconv/BUILD.bazel
--- c/go/analysis/passes/stringintconv/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/stringintconv/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,6 +12,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["string_test.go"],
diff -urN c/go/analysis/passes/structtag/BUILD.bazel d/go/analysis/passes/structtag/BUILD.bazel
--- c/go/analysis/passes/structtag/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/structtag/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,6 +12,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["structtag_test.go"],
diff -urN c/go/analysis/passes/testinggoroutine/BUILD.bazel d/go/analysis/passes/testinggoroutine/BUILD.bazel
--- c/go/analysis/passes/testinggoroutine/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/testinggoroutine/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["testinggoroutine_test.go"],
diff -urN c/go/analysis/passes/tests/BUILD.bazel d/go/analysis/passes/tests/BUILD.bazel
--- c/go/analysis/passes/tests/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/tests/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -8,6 +8,12 @@
     deps = ["//go/analysis:go_default_library"],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["tests_test.go"],
diff -urN c/go/analysis/passes/unmarshal/BUILD.bazel d/go/analysis/passes/unmarshal/BUILD.bazel
--- c/go/analysis/passes/unmarshal/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/unmarshal/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["unmarshal_test.go"],
diff -urN c/go/analysis/passes/unreachable/BUILD.bazel d/go/analysis/passes/unreachable/BUILD.bazel
--- c/go/analysis/passes/unreachable/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/unreachable/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,6 +12,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["unreachable_test.go"],
diff -urN c/go/analysis/passes/unsafeptr/BUILD.bazel d/go/analysis/passes/unsafeptr/BUILD.bazel
--- c/go/analysis/passes/unsafeptr/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/unsafeptr/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,6 +12,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["unsafeptr_test.go"],
diff -urN c/go/analysis/passes/unusedresult/BUILD.bazel d/go/analysis/passes/unusedresult/BUILD.bazel
--- c/go/analysis/passes/unusedresult/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/analysis/passes/unusedresult/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -13,6 +13,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["unusedresult_test.go"],
diff -urN c/go/ast/astutil/BUILD.bazel d/go/ast/astutil/BUILD.bazel
--- c/go/ast/astutil/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/ast/astutil/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,6 +12,12 @@
     visibility = ["//visibility:public"],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = [
diff -urN c/go/ast/inspector/BUILD.bazel d/go/ast/inspector/BUILD.bazel
--- c/go/ast/inspector/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/ast/inspector/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -10,6 +10,12 @@
     visibility = ["//visibility:public"],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["inspector_test.go"],
diff -urN c/go/cfg/BUILD.bazel d/go/cfg/BUILD.bazel
--- c/go/cfg/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/cfg/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -10,6 +10,12 @@
     visibility = ["//visibility:public"],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["cfg_test.go"],
diff -urN c/go/gcexportdata/BUILD.bazel d/go/gcexportdata/BUILD.bazel
--- c/go/gcexportdata/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/gcexportdata/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -11,6 +11,12 @@
     deps = ["//go/internal/gcimporter:go_default_library"],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = [
diff -urN c/go/internal/gcimporter/BUILD.bazel d/go/internal/gcimporter/BUILD.bazel
--- c/go/internal/gcimporter/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/internal/gcimporter/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -16,6 +16,12 @@
     visibility = ["//go:__subpackages__"],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = [
diff -urN c/go/ssa/BUILD.bazel d/go/ssa/BUILD.bazel
--- c/go/ssa/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/ssa/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -33,6 +33,12 @@
     ],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = [
diff -urN c/go/types/objectpath/BUILD.bazel d/go/types/objectpath/BUILD.bazel
--- c/go/types/objectpath/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/types/objectpath/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -7,6 +7,12 @@
     visibility = ["//visibility:public"],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["objectpath_test.go"],
diff -urN c/go/types/typeutil/BUILD.bazel d/go/types/typeutil/BUILD.bazel
--- c/go/types/typeutil/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ d/go/types/typeutil/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -14,6 +14,12 @@
     deps = ["//go/ast/astutil:go_default_library"],
 )
 
+alias(
+    name = "go_tool_library",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = [
