diff -urN a/balancer/base/BUILD.bazel b/balancer/base/BUILD.bazel
--- a/balancer/base/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/balancer/base/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,18 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "balancer.go",
+        "base.go",
+    ],
+    importpath = "google.golang.org/grpc/balancer/base",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//balancer:go_default_library",
+        "//connectivity:go_default_library",
+        "//grpclog:go_default_library",
+        "//resolver:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/balancer/BUILD.bazel b/balancer/BUILD.bazel
--- a/balancer/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/balancer/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,14 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["balancer.go"],
+    importpath = "google.golang.org/grpc/balancer",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//connectivity:go_default_library",
+        "//credentials:go_default_library",
+        "//resolver:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/balancer/grpclb/BUILD.bazel b/balancer/grpclb/BUILD.bazel
--- a/balancer/grpclb/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/balancer/grpclb/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,56 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "grpclb.go",
+        "grpclb_picker.go",
+        "grpclb_remote_balancer.go",
+        "grpclb_util.go",
+    ],
+    importpath = "google.golang.org/grpc/balancer/grpclb",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "//balancer:go_default_library",
+        "//balancer/grpclb/grpc_lb_v1:go_default_library",
+        "//codes:go_default_library",
+        "//connectivity:go_default_library",
+        "//grpclog:go_default_library",
+        "//internal:go_default_library",
+        "//internal/backoff:go_default_library",
+        "//internal/channelz:go_default_library",
+        "//metadata:go_default_library",
+        "//resolver:go_default_library",
+        "//status:go_default_library",
+        "@io_bazel_rules_go//proto/wkt:duration_go_proto",
+        "@io_bazel_rules_go//proto/wkt:timestamp_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "grpclb_test.go",
+        "grpclb_util_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = [
+        "//:go_default_library",
+        "//balancer:go_default_library",
+        "//balancer/grpclb/grpc_lb_v1:go_default_library",
+        "//codes:go_default_library",
+        "//credentials:go_default_library",
+        "//grpclog/glogger:go_default_library",
+        "//internal/leakcheck:go_default_library",
+        "//metadata:go_default_library",
+        "//peer:go_default_library",
+        "//resolver:go_default_library",
+        "//resolver/manual:go_default_library",
+        "//status:go_default_library",
+        "//test/grpc_testing:go_default_library",
+        "@io_bazel_rules_go//proto/wkt:duration_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/balancer/grpclb/grpc_lb_v1/BUILD.bazel b/balancer/grpclb/grpc_lb_v1/BUILD.bazel
--- a/balancer/grpclb/grpc_lb_v1/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/balancer/grpclb/grpc_lb_v1/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,15 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["load_balancer.pb.go"],
+    importpath = "google.golang.org/grpc/balancer/grpclb/grpc_lb_v1",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@io_bazel_rules_go//proto/wkt:duration_go_proto",
+        "@io_bazel_rules_go//proto/wkt:timestamp_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/balancer/roundrobin/BUILD.bazel b/balancer/roundrobin/BUILD.bazel
--- a/balancer/roundrobin/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/balancer/roundrobin/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,33 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["roundrobin.go"],
+    importpath = "google.golang.org/grpc/balancer/roundrobin",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//balancer:go_default_library",
+        "//balancer/base:go_default_library",
+        "//grpclog:go_default_library",
+        "//resolver:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["roundrobin_test.go"],
+    embed = [":go_default_library"],
+    deps = [
+        "//:go_default_library",
+        "//codes:go_default_library",
+        "//grpclog/glogger:go_default_library",
+        "//internal/leakcheck:go_default_library",
+        "//peer:go_default_library",
+        "//resolver:go_default_library",
+        "//resolver/manual:go_default_library",
+        "//status:go_default_library",
+        "//test/grpc_testing:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/benchmark/benchmain/BUILD.bazel b/benchmark/benchmain/BUILD.bazel
--- a/benchmark/benchmain/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/benchmark/benchmain/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,24 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/benchmark/benchmain",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//benchmark:go_default_library",
+        "//benchmark/grpc_testing:go_default_library",
+        "//benchmark/latency:go_default_library",
+        "//benchmark/stats:go_default_library",
+        "//grpclog:go_default_library",
+        "//test/bufconn:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "benchmain",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/benchmark/benchresult/BUILD.bazel b/benchmark/benchresult/BUILD.bazel
--- a/benchmark/benchresult/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/benchmark/benchresult/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,15 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/benchmark/benchresult",
+    visibility = ["//visibility:private"],
+    deps = ["//benchmark/stats:go_default_library"],
+)
+
+go_binary(
+    name = "benchresult",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/benchmark/BUILD.bazel b/benchmark/BUILD.bazel
--- a/benchmark/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/benchmark/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,29 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["benchmark.go"],
+    importpath = "google.golang.org/grpc/benchmark",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "//benchmark/grpc_testing:go_default_library",
+        "//benchmark/latency:go_default_library",
+        "//benchmark/stats:go_default_library",
+        "//grpclog:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "benchmark16_test.go",
+        "benchmark17_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = [
+        "//:go_default_library",
+        "//benchmark/stats:go_default_library",
+    ],
+)
diff -urN a/benchmark/client/BUILD.bazel b/benchmark/client/BUILD.bazel
--- a/benchmark/client/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/benchmark/client/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,23 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/benchmark/client",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//benchmark:go_default_library",
+        "//benchmark/grpc_testing:go_default_library",
+        "//benchmark/stats:go_default_library",
+        "//grpclog:go_default_library",
+        "//internal/syscall:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "client",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/benchmark/grpc_testing/BUILD.bazel b/benchmark/grpc_testing/BUILD.bazel
--- a/benchmark/grpc_testing/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/benchmark/grpc_testing/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,19 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "control.pb.go",
+        "messages.pb.go",
+        "payloads.pb.go",
+        "services.pb.go",
+        "stats.pb.go",
+    ],
+    importpath = "google.golang.org/grpc/benchmark/grpc_testing",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/benchmark/latency/BUILD.bazel b/benchmark/latency/BUILD.bazel
--- a/benchmark/latency/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/benchmark/latency/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,15 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["latency.go"],
+    importpath = "google.golang.org/grpc/benchmark/latency",
+    visibility = ["//visibility:public"],
+    deps = ["@org_golang_x_net//context:go_default_library"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["latency_test.go"],
+    embed = [":go_default_library"],
+)
diff -urN a/benchmark/primitives/BUILD.bazel b/benchmark/primitives/BUILD.bazel
--- a/benchmark/primitives/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/benchmark/primitives/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,14 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "code_string_test.go",
+        "context_test.go",
+        "primitives_test.go",
+    ],
+    deps = [
+        "//codes:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/benchmark/server/BUILD.bazel b/benchmark/server/BUILD.bazel
--- a/benchmark/server/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/benchmark/server/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,19 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/benchmark/server",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//benchmark:go_default_library",
+        "//grpclog:go_default_library",
+        "//internal/syscall:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "server",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/benchmark/stats/BUILD.bazel b/benchmark/stats/BUILD.bazel
--- a/benchmark/stats/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/benchmark/stats/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,12 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "histogram.go",
+        "stats.go",
+        "util.go",
+    ],
+    importpath = "google.golang.org/grpc/benchmark/stats",
+    visibility = ["//visibility:public"],
+)
diff -urN a/benchmark/worker/BUILD.bazel b/benchmark/worker/BUILD.bazel
--- a/benchmark/worker/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/benchmark/worker/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,31 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "benchmark_client.go",
+        "benchmark_server.go",
+        "main.go",
+    ],
+    importpath = "google.golang.org/grpc/benchmark/worker",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//benchmark:go_default_library",
+        "//benchmark/grpc_testing:go_default_library",
+        "//benchmark/stats:go_default_library",
+        "//codes:go_default_library",
+        "//credentials:go_default_library",
+        "//grpclog:go_default_library",
+        "//internal/syscall:go_default_library",
+        "//status:go_default_library",
+        "//testdata:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "worker",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/BUILD.bazel	2018-09-27 16:55:16.114787111 -0400
@@ -0,0 +1,103 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "backoff.go",
+        "balancer.go",
+        "balancer_conn_wrappers.go",
+        "balancer_v1_wrapper.go",
+        "call.go",
+        "clientconn.go",
+        "codec.go",
+        "dialoptions.go",
+        "doc.go",
+        "go16.go",
+        "go17.go",
+        "interceptor.go",
+        "picker_wrapper.go",
+        "pickfirst.go",
+        "proxy.go",
+        "resolver_conn_wrapper.go",
+        "rpc_util.go",
+        "server.go",
+        "service_config.go",
+        "stream.go",
+        "trace.go",
+        "version.go",
+    ],
+    importpath = "google.golang.org/grpc",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//balancer:go_default_library",
+        "//balancer/roundrobin:go_default_library",
+        "//codes:go_default_library",
+        "//connectivity:go_default_library",
+        "//credentials:go_default_library",
+        "//encoding:go_default_library",
+        "//encoding/proto:go_default_library",
+        "//grpclog:go_default_library",
+        "//internal:go_default_library",
+        "//internal/backoff:go_default_library",
+        "//internal/channelz:go_default_library",
+        "//internal/envconfig:go_default_library",
+        "//internal/grpcrand:go_default_library",
+        "//internal/transport:go_default_library",
+        "//keepalive:go_default_library",
+        "//metadata:go_default_library",
+        "//naming:go_default_library",
+        "//peer:go_default_library",
+        "//resolver:go_default_library",
+        "//resolver/dns:go_default_library",
+        "//resolver/passthrough:go_default_library",
+        "//stats:go_default_library",
+        "//status:go_default_library",
+        "//tap:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+        "@org_golang_x_net//http2:go_default_library",
+        "@org_golang_x_net//trace:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "balancer_switching_test.go",
+        "balancer_test.go",
+        "call_test.go",
+        "clientconn_test.go",
+        "codec_test.go",
+        "picker_wrapper_test.go",
+        "pickfirst_test.go",
+        "proxy_test.go",
+        "resolver_conn_wrapper_test.go",
+        "rpc_util_test.go",
+        "server_test.go",
+        "service_config_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = [
+        "//balancer:go_default_library",
+        "//balancer/roundrobin:go_default_library",
+        "//codes:go_default_library",
+        "//connectivity:go_default_library",
+        "//credentials:go_default_library",
+        "//encoding:go_default_library",
+        "//encoding/proto:go_default_library",
+        "//grpclog/glogger:go_default_library",
+        "//internal/backoff:go_default_library",
+        "//internal/leakcheck:go_default_library",
+        "//internal/transport:go_default_library",
+        "//keepalive:go_default_library",
+        "//naming:go_default_library",
+        "//resolver:go_default_library",
+        "//resolver/manual:go_default_library",
+        "//resolver/passthrough:go_default_library",
+        "//status:go_default_library",
+        "//test/codec_perf:go_default_library",
+        "//testdata:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+        "@org_golang_x_net//http2:go_default_library",
+    ],
+)
diff -urN a/channelz/grpc_channelz_v1/BUILD.bazel b/channelz/grpc_channelz_v1/BUILD.bazel
--- a/channelz/grpc_channelz_v1/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/channelz/grpc_channelz_v1/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,17 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["channelz.pb.go"],
+    importpath = "google.golang.org/grpc/channelz/grpc_channelz_v1",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@io_bazel_rules_go//proto/wkt:any_go_proto",
+        "@io_bazel_rules_go//proto/wkt:duration_go_proto",
+        "@io_bazel_rules_go//proto/wkt:timestamp_go_proto",
+        "@io_bazel_rules_go//proto/wkt:wrappers_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/channelz/service/BUILD.bazel b/channelz/service/BUILD.bazel
--- a/channelz/service/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/channelz/service/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,53 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "func_linux.go",
+        "func_nonlinux.go",
+        "service.go",
+    ],
+    importpath = "google.golang.org/grpc/channelz/service",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "//channelz/grpc_channelz_v1:go_default_library",
+        "//connectivity:go_default_library",
+        "//credentials:go_default_library",
+        "//internal/channelz:go_default_library",
+        "@com_github_golang_protobuf//ptypes:go_default_library_gen",
+        "@io_bazel_rules_go//proto/wkt:duration_go_proto",
+        "@io_bazel_rules_go//proto/wkt:wrappers_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "service_sktopt_test.go",
+        "service_test.go",
+        "util_sktopt_386_test.go",
+        "util_sktopt_amd64_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = [
+        "//channelz/grpc_channelz_v1:go_default_library",
+        "//connectivity:go_default_library",
+        "//credentials:go_default_library",
+        "//internal/channelz:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@com_github_golang_protobuf//ptypes:go_default_library_gen",
+        "@org_golang_x_net//context:go_default_library",
+    ] + select({
+        "@io_bazel_rules_go//go/platform:linux_386": [
+            "@io_bazel_rules_go//proto/wkt:duration_go_proto",
+            "@org_golang_x_sys//unix:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:linux_amd64": [
+            "@io_bazel_rules_go//proto/wkt:duration_go_proto",
+            "@org_golang_x_sys//unix:go_default_library",
+        ],
+        "//conditions:default": [],
+    }),
+)
diff -urN a/codes/BUILD.bazel b/codes/BUILD.bazel
--- a/codes/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/codes/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,18 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "code_string.go",
+        "codes.go",
+    ],
+    importpath = "google.golang.org/grpc/codes",
+    visibility = ["//visibility:public"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["codes_test.go"],
+    embed = [":go_default_library"],
+    deps = ["@go_googleapis//google/rpc:code_go_proto"],
+)
diff -urN a/connectivity/BUILD.bazel b/connectivity/BUILD.bazel
--- a/connectivity/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/connectivity/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,12 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["connectivity.go"],
+    importpath = "google.golang.org/grpc/connectivity",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//grpclog:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/credentials/alts/BUILD.bazel b/credentials/alts/BUILD.bazel
--- a/credentials/alts/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/credentials/alts/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,36 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "alts.go",
+        "utils.go",
+    ],
+    importpath = "google.golang.org/grpc/credentials/alts",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//credentials:go_default_library",
+        "//credentials/alts/core:go_default_library",
+        "//credentials/alts/core/handshaker:go_default_library",
+        "//credentials/alts/core/handshaker/service:go_default_library",
+        "//credentials/alts/core/proto/grpc_gcp:go_default_library",
+        "//grpclog:go_default_library",
+        "//peer:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "alts_test.go",
+        "utils_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = [
+        "//credentials/alts/core/proto/grpc_gcp:go_default_library",
+        "//peer:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/credentials/alts/core/authinfo/BUILD.bazel b/credentials/alts/core/authinfo/BUILD.bazel
--- a/credentials/alts/core/authinfo/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/credentials/alts/core/authinfo/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,19 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["authinfo.go"],
+    importpath = "google.golang.org/grpc/credentials/alts/core/authinfo",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//credentials:go_default_library",
+        "//credentials/alts/core/proto/grpc_gcp:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["authinfo_test.go"],
+    embed = [":go_default_library"],
+    deps = ["//credentials/alts/core/proto/grpc_gcp:go_default_library"],
+)
diff -urN a/credentials/alts/core/BUILD.bazel b/credentials/alts/core/BUILD.bazel
--- a/credentials/alts/core/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/credentials/alts/core/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,12 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["common.go"],
+    importpath = "google.golang.org/grpc/credentials/alts/core",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//credentials:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/credentials/alts/core/conn/BUILD.bazel b/credentials/alts/core/conn/BUILD.bazel
--- a/credentials/alts/core/conn/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/credentials/alts/core/conn/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,29 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "aeadrekey.go",
+        "aes128gcm.go",
+        "aes128gcmrekey.go",
+        "common.go",
+        "counter.go",
+        "record.go",
+    ],
+    importpath = "google.golang.org/grpc/credentials/alts/core/conn",
+    visibility = ["//visibility:public"],
+    deps = ["//credentials/alts/core:go_default_library"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "aeadrekey_test.go",
+        "aes128gcm_test.go",
+        "aes128gcmrekey_test.go",
+        "counter_test.go",
+        "record_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = ["//credentials/alts/core:go_default_library"],
+)
diff -urN a/credentials/alts/core/handshaker/BUILD.bazel b/credentials/alts/core/handshaker/BUILD.bazel
--- a/credentials/alts/core/handshaker/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/credentials/alts/core/handshaker/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,31 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["handshaker.go"],
+    importpath = "google.golang.org/grpc/credentials/alts/core/handshaker",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "//codes:go_default_library",
+        "//credentials:go_default_library",
+        "//credentials/alts/core:go_default_library",
+        "//credentials/alts/core/authinfo:go_default_library",
+        "//credentials/alts/core/conn:go_default_library",
+        "//credentials/alts/core/proto/grpc_gcp:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["handshaker_test.go"],
+    embed = [":go_default_library"],
+    deps = [
+        "//:go_default_library",
+        "//credentials/alts/core:go_default_library",
+        "//credentials/alts/core/proto/grpc_gcp:go_default_library",
+        "//credentials/alts/core/testutil:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/credentials/alts/core/handshaker/service/BUILD.bazel b/credentials/alts/core/handshaker/service/BUILD.bazel
--- a/credentials/alts/core/handshaker/service/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/credentials/alts/core/handshaker/service/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,16 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["service.go"],
+    importpath = "google.golang.org/grpc/credentials/alts/core/handshaker/service",
+    visibility = ["//visibility:public"],
+    deps = ["//:go_default_library"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["service_test.go"],
+    embed = [":go_default_library"],
+    deps = ["//:go_default_library"],
+)
diff -urN a/credentials/alts/core/proto/grpc_gcp/BUILD.bazel b/credentials/alts/core/proto/grpc_gcp/BUILD.bazel
--- a/credentials/alts/core/proto/grpc_gcp/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/credentials/alts/core/proto/grpc_gcp/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,17 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "altscontext.pb.go",
+        "handshaker.pb.go",
+        "transport_security_common.pb.go",
+    ],
+    importpath = "google.golang.org/grpc/credentials/alts/core/proto/grpc_gcp",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/credentials/alts/core/testutil/BUILD.bazel b/credentials/alts/core/testutil/BUILD.bazel
--- a/credentials/alts/core/testutil/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/credentials/alts/core/testutil/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,9 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["testutil.go"],
+    importpath = "google.golang.org/grpc/credentials/alts/core/testutil",
+    visibility = ["//visibility:public"],
+    deps = ["//credentials/alts/core/conn:go_default_library"],
+)
diff -urN a/credentials/BUILD.bazel b/credentials/BUILD.bazel
--- a/credentials/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/credentials/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,28 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "credentials.go",
+        "go16.go",
+        "go17.go",
+        "go18.go",
+        "go19.go",
+    ],
+    importpath = "google.golang.org/grpc/credentials",
+    visibility = ["//visibility:public"],
+    deps = [
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["credentials_test.go"],
+    embed = [":go_default_library"],
+    deps = [
+        "//testdata:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/credentials/oauth/BUILD.bazel b/credentials/oauth/BUILD.bazel
--- a/credentials/oauth/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/credentials/oauth/BUILD.bazel	2018-09-27 16:55:16.106787007 -0400
@@ -0,0 +1,15 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["oauth.go"],
+    importpath = "google.golang.org/grpc/credentials/oauth",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//credentials:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+        "@org_golang_x_oauth2//:go_default_library",
+        "@org_golang_x_oauth2//google:go_default_library",
+        "@org_golang_x_oauth2//jwt:go_default_library",
+    ],
+)
diff -urN a/encoding/BUILD.bazel b/encoding/BUILD.bazel
--- a/encoding/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/encoding/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,8 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["encoding.go"],
+    importpath = "google.golang.org/grpc/encoding",
+    visibility = ["//visibility:public"],
+)
diff -urN a/encoding/gzip/BUILD.bazel b/encoding/gzip/BUILD.bazel
--- a/encoding/gzip/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/encoding/gzip/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,9 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["gzip.go"],
+    importpath = "google.golang.org/grpc/encoding/gzip",
+    visibility = ["//visibility:public"],
+    deps = ["//encoding:go_default_library"],
+)
diff -urN a/encoding/proto/BUILD.bazel b/encoding/proto/BUILD.bazel
--- a/encoding/proto/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/encoding/proto/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,26 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["proto.go"],
+    importpath = "google.golang.org/grpc/encoding/proto",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//encoding:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "proto_benchmark_test.go",
+        "proto_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = [
+        "//encoding:go_default_library",
+        "//test/codec_perf:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+    ],
+)
diff -urN a/examples/helloworld/greeter_client/BUILD.bazel b/examples/helloworld/greeter_client/BUILD.bazel
--- a/examples/helloworld/greeter_client/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/helloworld/greeter_client/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,19 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/examples/helloworld/greeter_client",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//examples/helloworld/helloworld:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "greeter_client",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/examples/helloworld/greeter_server/BUILD.bazel b/examples/helloworld/greeter_server/BUILD.bazel
--- a/examples/helloworld/greeter_server/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/helloworld/greeter_server/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,20 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/examples/helloworld/greeter_server",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//examples/helloworld/helloworld:go_default_library",
+        "//reflection:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "greeter_server",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/examples/helloworld/helloworld/BUILD.bazel b/examples/helloworld/helloworld/BUILD.bazel
--- a/examples/helloworld/helloworld/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/helloworld/helloworld/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,13 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["helloworld.pb.go"],
+    importpath = "google.golang.org/grpc/examples/helloworld/helloworld",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/examples/helloworld/mock_helloworld/BUILD.bazel b/examples/helloworld/mock_helloworld/BUILD.bazel
--- a/examples/helloworld/mock_helloworld/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/helloworld/mock_helloworld/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,26 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["hw_mock.go"],
+    importpath = "google.golang.org/grpc/examples/helloworld/mock_helloworld",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "//examples/helloworld/helloworld:go_default_library",
+        "@com_github_golang_mock//gomock:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["hw_mock_test.go"],
+    embed = [":go_default_library"],
+    deps = [
+        "//examples/helloworld/helloworld:go_default_library",
+        "@com_github_golang_mock//gomock:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/examples/oauth/client/BUILD.bazel b/examples/oauth/client/BUILD.bazel
--- a/examples/oauth/client/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/oauth/client/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,22 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/examples/oauth/client",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//credentials:go_default_library",
+        "//credentials/oauth:go_default_library",
+        "//examples/helloworld/helloworld:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+        "@org_golang_x_oauth2//:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "client",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/examples/oauth/server/BUILD.bazel b/examples/oauth/server/BUILD.bazel
--- a/examples/oauth/server/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/oauth/server/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,24 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/examples/oauth/server",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//codes:go_default_library",
+        "//credentials:go_default_library",
+        "//examples/helloworld/helloworld:go_default_library",
+        "//metadata:go_default_library",
+        "//status:go_default_library",
+        "//testdata:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "server",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/examples/route_guide/client/BUILD.bazel b/examples/route_guide/client/BUILD.bazel
--- a/examples/route_guide/client/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/route_guide/client/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,21 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["client.go"],
+    importpath = "google.golang.org/grpc/examples/route_guide/client",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//credentials:go_default_library",
+        "//examples/route_guide/routeguide:go_default_library",
+        "//testdata:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "client",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/examples/route_guide/mock_routeguide/BUILD.bazel b/examples/route_guide/mock_routeguide/BUILD.bazel
--- a/examples/route_guide/mock_routeguide/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/route_guide/mock_routeguide/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,27 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["rg_mock.go"],
+    importpath = "google.golang.org/grpc/examples/route_guide/mock_routeguide",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "//examples/route_guide/routeguide:go_default_library",
+        "//metadata:go_default_library",
+        "@com_github_golang_mock//gomock:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["rg_mock_test.go"],
+    embed = [":go_default_library"],
+    deps = [
+        "//examples/route_guide/routeguide:go_default_library",
+        "@com_github_golang_mock//gomock:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/examples/route_guide/routeguide/BUILD.bazel b/examples/route_guide/routeguide/BUILD.bazel
--- a/examples/route_guide/routeguide/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/route_guide/routeguide/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,13 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["route_guide.pb.go"],
+    importpath = "google.golang.org/grpc/examples/route_guide/routeguide",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/examples/route_guide/server/BUILD.bazel b/examples/route_guide/server/BUILD.bazel
--- a/examples/route_guide/server/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/route_guide/server/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,22 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["server.go"],
+    importpath = "google.golang.org/grpc/examples/route_guide/server",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//credentials:go_default_library",
+        "//examples/route_guide/routeguide:go_default_library",
+        "//testdata:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "server",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/examples/rpc_errors/client/BUILD.bazel b/examples/rpc_errors/client/BUILD.bazel
--- a/examples/rpc_errors/client/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/rpc_errors/client/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,21 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/examples/rpc_errors/client",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//examples/helloworld/helloworld:go_default_library",
+        "//status:go_default_library",
+        "@go_googleapis//google/rpc:errdetails_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "client",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/examples/rpc_errors/server/BUILD.bazel b/examples/rpc_errors/server/BUILD.bazel
--- a/examples/rpc_errors/server/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/rpc_errors/server/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,22 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/examples/rpc_errors/server",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//codes:go_default_library",
+        "//examples/helloworld/helloworld:go_default_library",
+        "//status:go_default_library",
+        "@go_googleapis//google/rpc:errdetails_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "server",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/grpclog/BUILD.bazel b/grpclog/BUILD.bazel
--- a/grpclog/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/grpclog/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,18 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "grpclog.go",
+        "logger.go",
+        "loggerv2.go",
+    ],
+    importpath = "google.golang.org/grpc/grpclog",
+    visibility = ["//visibility:public"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["loggerv2_test.go"],
+    embed = [":go_default_library"],
+)
diff -urN a/grpclog/glogger/BUILD.bazel b/grpclog/glogger/BUILD.bazel
--- a/grpclog/glogger/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/grpclog/glogger/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,12 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["glogger.go"],
+    importpath = "google.golang.org/grpc/grpclog/glogger",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//grpclog:go_default_library",
+        "@com_github_golang_glog//:go_default_library",
+    ],
+)
diff -urN a/health/BUILD.bazel b/health/BUILD.bazel
--- a/health/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/health/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,14 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["health.go"],
+    importpath = "google.golang.org/grpc/health",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//codes:go_default_library",
+        "//health/grpc_health_v1:go_default_library",
+        "//status:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/health/grpc_health_v1/BUILD.bazel b/health/grpc_health_v1/BUILD.bazel
--- a/health/grpc_health_v1/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/health/grpc_health_v1/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,13 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["health.pb.go"],
+    importpath = "google.golang.org/grpc/health/grpc_health_v1",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/internal/backoff/BUILD.bazel b/internal/backoff/BUILD.bazel
--- a/internal/backoff/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/internal/backoff/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,9 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["backoff.go"],
+    importpath = "google.golang.org/grpc/internal/backoff",
+    visibility = ["//:__subpackages__"],
+    deps = ["//internal/grpcrand:go_default_library"],
+)
diff -urN a/internal/BUILD.bazel b/internal/BUILD.bazel
--- a/internal/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/internal/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,8 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["internal.go"],
+    importpath = "google.golang.org/grpc/internal",
+    visibility = ["//visibility:public"],
+)
diff -urN a/internal/channelz/BUILD.bazel b/internal/channelz/BUILD.bazel
--- a/internal/channelz/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/internal/channelz/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,37 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "funcs.go",
+        "types.go",
+        "types_linux.go",
+        "types_nonlinux.go",
+        "util_linux_go19.go",
+        "util_nonlinux_pre_go19.go",
+    ],
+    importpath = "google.golang.org/grpc/internal/channelz",
+    visibility = ["//:__subpackages__"],
+    deps = [
+        "//connectivity:go_default_library",
+        "//credentials:go_default_library",
+        "//grpclog:go_default_library",
+    ] + select({
+        "@io_bazel_rules_go//go/platform:linux": [
+            "@org_golang_x_sys//unix:go_default_library",
+        ],
+        "//conditions:default": [],
+    }),
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["util_test.go"],
+    embed = [":go_default_library"],
+    deps = select({
+        "@io_bazel_rules_go//go/platform:linux": [
+            "@org_golang_x_sys//unix:go_default_library",
+        ],
+        "//conditions:default": [],
+    }),
+)
diff -urN a/internal/envconfig/BUILD.bazel b/internal/envconfig/BUILD.bazel
--- a/internal/envconfig/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/internal/envconfig/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,8 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["envconfig.go"],
+    importpath = "google.golang.org/grpc/internal/envconfig",
+    visibility = ["//:__subpackages__"],
+)
diff -urN a/internal/grpcrand/BUILD.bazel b/internal/grpcrand/BUILD.bazel
--- a/internal/grpcrand/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/internal/grpcrand/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,8 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["grpcrand.go"],
+    importpath = "google.golang.org/grpc/internal/grpcrand",
+    visibility = ["//:__subpackages__"],
+)
diff -urN a/internal/grpcsync/BUILD.bazel b/internal/grpcsync/BUILD.bazel
--- a/internal/grpcsync/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/internal/grpcsync/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,14 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["event.go"],
+    importpath = "google.golang.org/grpc/internal/grpcsync",
+    visibility = ["//:__subpackages__"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["event_test.go"],
+    embed = [":go_default_library"],
+)
diff -urN a/internal/leakcheck/BUILD.bazel b/internal/leakcheck/BUILD.bazel
--- a/internal/leakcheck/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/internal/leakcheck/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,14 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["leakcheck.go"],
+    importpath = "google.golang.org/grpc/internal/leakcheck",
+    visibility = ["//:__subpackages__"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["leakcheck_test.go"],
+    embed = [":go_default_library"],
+)
diff -urN a/internal/syscall/BUILD.bazel b/internal/syscall/BUILD.bazel
--- a/internal/syscall/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/internal/syscall/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,48 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "syscall_linux.go",
+        "syscall_nonlinux.go",
+    ],
+    importpath = "google.golang.org/grpc/internal/syscall",
+    visibility = ["//:__subpackages__"],
+    deps = select({
+        "@io_bazel_rules_go//go/platform:android": [
+            "//grpclog:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:darwin": [
+            "//grpclog:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:dragonfly": [
+            "//grpclog:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:freebsd": [
+            "//grpclog:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:linux": [
+            "//grpclog:go_default_library",
+            "@org_golang_x_sys//unix:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:nacl": [
+            "//grpclog:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:netbsd": [
+            "//grpclog:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:openbsd": [
+            "//grpclog:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:plan9": [
+            "//grpclog:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:solaris": [
+            "//grpclog:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:windows": [
+            "//grpclog:go_default_library",
+        ],
+        "//conditions:default": [],
+    }),
+)
diff -urN a/internal/transport/BUILD.bazel b/internal/transport/BUILD.bazel
--- a/internal/transport/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/internal/transport/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,62 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "bdp_estimator.go",
+        "controlbuf.go",
+        "defaults.go",
+        "flowcontrol.go",
+        "go16.go",
+        "go17.go",
+        "handler_server.go",
+        "http2_client.go",
+        "http2_server.go",
+        "http_util.go",
+        "log.go",
+        "transport.go",
+    ],
+    importpath = "google.golang.org/grpc/internal/transport",
+    visibility = ["//:__subpackages__"],
+    deps = [
+        "//codes:go_default_library",
+        "//credentials:go_default_library",
+        "//grpclog:go_default_library",
+        "//internal/channelz:go_default_library",
+        "//internal/grpcrand:go_default_library",
+        "//keepalive:go_default_library",
+        "//metadata:go_default_library",
+        "//peer:go_default_library",
+        "//stats:go_default_library",
+        "//status:go_default_library",
+        "//tap:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@go_googleapis//google/rpc:status_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+        "@org_golang_x_net//http2:go_default_library",
+        "@org_golang_x_net//http2/hpack:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "handler_server_test.go",
+        "http_util_test.go",
+        "transport_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = [
+        "//codes:go_default_library",
+        "//internal/leakcheck:go_default_library",
+        "//keepalive:go_default_library",
+        "//metadata:go_default_library",
+        "//status:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@go_googleapis//google/rpc:errdetails_go_proto",
+        "@io_bazel_rules_go//proto/wkt:duration_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+        "@org_golang_x_net//http2:go_default_library",
+        "@org_golang_x_net//http2/hpack:go_default_library",
+    ],
+)
diff -urN a/interop/alts/client/BUILD.bazel b/interop/alts/client/BUILD.bazel
--- a/interop/alts/client/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/interop/alts/client/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,21 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["client.go"],
+    importpath = "google.golang.org/grpc/interop/alts/client",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//credentials/alts:go_default_library",
+        "//grpclog:go_default_library",
+        "//interop/grpc_testing:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "client",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/interop/alts/server/BUILD.bazel b/interop/alts/server/BUILD.bazel
--- a/interop/alts/server/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/interop/alts/server/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,21 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["server.go"],
+    importpath = "google.golang.org/grpc/interop/alts/server",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//credentials/alts:go_default_library",
+        "//grpclog:go_default_library",
+        "//interop:go_default_library",
+        "//interop/grpc_testing:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "server",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/interop/BUILD.bazel b/interop/BUILD.bazel
--- a/interop/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/interop/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,20 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["test_utils.go"],
+    importpath = "google.golang.org/grpc/interop",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "//codes:go_default_library",
+        "//grpclog:go_default_library",
+        "//interop/grpc_testing:go_default_library",
+        "//metadata:go_default_library",
+        "//status:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+        "@org_golang_x_oauth2//:go_default_library",
+        "@org_golang_x_oauth2//google:go_default_library",
+    ],
+)
diff -urN a/interop/client/BUILD.bazel b/interop/client/BUILD.bazel
--- a/interop/client/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/interop/client/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,26 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["client.go"],
+    importpath = "google.golang.org/grpc/interop/client",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//balancer/grpclb:go_default_library",
+        "//credentials:go_default_library",
+        "//credentials/alts:go_default_library",
+        "//credentials/oauth:go_default_library",
+        "//grpclog:go_default_library",
+        "//interop:go_default_library",
+        "//interop/grpc_testing:go_default_library",
+        "//resolver:go_default_library",
+        "//testdata:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "client",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/interop/grpc_testing/BUILD.bazel b/interop/grpc_testing/BUILD.bazel
--- a/interop/grpc_testing/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/interop/grpc_testing/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,13 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["test.pb.go"],
+    importpath = "google.golang.org/grpc/interop/grpc_testing",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/interop/http2/BUILD.bazel b/interop/http2/BUILD.bazel
--- a/interop/http2/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/interop/http2/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,23 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["negative_http2_client.go"],
+    importpath = "google.golang.org/grpc/interop/http2",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//codes:go_default_library",
+        "//grpclog:go_default_library",
+        "//interop:go_default_library",
+        "//interop/grpc_testing:go_default_library",
+        "//status:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "http2",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/interop/server/BUILD.bazel b/interop/server/BUILD.bazel
--- a/interop/server/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/interop/server/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,23 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["server.go"],
+    importpath = "google.golang.org/grpc/interop/server",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//credentials:go_default_library",
+        "//credentials/alts:go_default_library",
+        "//grpclog:go_default_library",
+        "//interop:go_default_library",
+        "//interop/grpc_testing:go_default_library",
+        "//testdata:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "server",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/keepalive/BUILD.bazel b/keepalive/BUILD.bazel
--- a/keepalive/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/keepalive/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,8 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["keepalive.go"],
+    importpath = "google.golang.org/grpc/keepalive",
+    visibility = ["//visibility:public"],
+)
diff -urN a/metadata/BUILD.bazel b/metadata/BUILD.bazel
--- a/metadata/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/metadata/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,16 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["metadata.go"],
+    importpath = "google.golang.org/grpc/metadata",
+    visibility = ["//visibility:public"],
+    deps = ["@org_golang_x_net//context:go_default_library"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["metadata_test.go"],
+    embed = [":go_default_library"],
+    deps = ["@org_golang_x_net//context:go_default_library"],
+)
diff -urN a/naming/BUILD.bazel b/naming/BUILD.bazel
--- a/naming/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/naming/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,28 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "dns_resolver.go",
+        "go17.go",
+        "go18.go",
+        "naming.go",
+    ],
+    importpath = "google.golang.org/grpc/naming",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//grpclog:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "dns_resolver_test.go",
+        "go17_test.go",
+        "go18_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = ["@org_golang_x_net//context:go_default_library"],
+)
diff -urN a/peer/BUILD.bazel b/peer/BUILD.bazel
--- a/peer/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/peer/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,12 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["peer.go"],
+    importpath = "google.golang.org/grpc/peer",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//credentials:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/reflection/BUILD.bazel b/reflection/BUILD.bazel
--- a/reflection/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/reflection/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,31 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["serverreflection.go"],
+    importpath = "google.golang.org/grpc/reflection",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "//codes:go_default_library",
+        "//reflection/grpc_reflection_v1alpha:go_default_library",
+        "//status:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@io_bazel_rules_go//proto/wkt:descriptor_go_proto",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["serverreflection_test.go"],
+    embed = [":go_default_library"],
+    deps = [
+        "//:go_default_library",
+        "//reflection/grpc_reflection_v1alpha:go_default_library",
+        "//reflection/grpc_testing:go_default_library",
+        "//reflection/grpc_testingv3:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@io_bazel_rules_go//proto/wkt:descriptor_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/reflection/grpc_reflection_v1alpha/BUILD.bazel b/reflection/grpc_reflection_v1alpha/BUILD.bazel
--- a/reflection/grpc_reflection_v1alpha/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/reflection/grpc_reflection_v1alpha/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,13 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["reflection.pb.go"],
+    importpath = "google.golang.org/grpc/reflection/grpc_reflection_v1alpha",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/reflection/grpc_testing/BUILD.bazel b/reflection/grpc_testing/BUILD.bazel
--- a/reflection/grpc_testing/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/reflection/grpc_testing/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,18 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "proto2.pb.go",
+        "proto2_ext.pb.go",
+        "proto2_ext2.pb.go",
+        "test.pb.go",
+    ],
+    importpath = "google.golang.org/grpc/reflection/grpc_testing",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/reflection/grpc_testingv3/BUILD.bazel b/reflection/grpc_testingv3/BUILD.bazel
--- a/reflection/grpc_testingv3/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/reflection/grpc_testingv3/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,13 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["testv3.pb.go"],
+    importpath = "google.golang.org/grpc/reflection/grpc_testingv3",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/resolver/BUILD.bazel b/resolver/BUILD.bazel
--- a/resolver/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/resolver/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,8 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["resolver.go"],
+    importpath = "google.golang.org/grpc/resolver",
+    visibility = ["//visibility:public"],
+)
diff -urN a/resolver/dns/BUILD.bazel b/resolver/dns/BUILD.bazel
--- a/resolver/dns/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/resolver/dns/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,34 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "dns_resolver.go",
+        "go17.go",
+        "go18.go",
+    ],
+    importpath = "google.golang.org/grpc/resolver/dns",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//grpclog:go_default_library",
+        "//internal/backoff:go_default_library",
+        "//internal/grpcrand:go_default_library",
+        "//resolver:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "dns_resolver_test.go",
+        "go17_test.go",
+        "go18_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = [
+        "//internal/leakcheck:go_default_library",
+        "//resolver:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/resolver/manual/BUILD.bazel b/resolver/manual/BUILD.bazel
--- a/resolver/manual/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/resolver/manual/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,9 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["manual.go"],
+    importpath = "google.golang.org/grpc/resolver/manual",
+    visibility = ["//visibility:public"],
+    deps = ["//resolver:go_default_library"],
+)
diff -urN a/resolver/passthrough/BUILD.bazel b/resolver/passthrough/BUILD.bazel
--- a/resolver/passthrough/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/resolver/passthrough/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,9 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["passthrough.go"],
+    importpath = "google.golang.org/grpc/resolver/passthrough",
+    visibility = ["//visibility:public"],
+    deps = ["//resolver:go_default_library"],
+)
diff -urN a/stats/BUILD.bazel b/stats/BUILD.bazel
--- a/stats/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/stats/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,26 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "handlers.go",
+        "stats.go",
+    ],
+    importpath = "google.golang.org/grpc/stats",
+    visibility = ["//visibility:public"],
+    deps = ["@org_golang_x_net//context:go_default_library"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["stats_test.go"],
+    embed = [":go_default_library"],
+    deps = [
+        "//:go_default_library",
+        "//metadata:go_default_library",
+        "//stats/grpc_testing:go_default_library",
+        "//status:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/stats/grpc_testing/BUILD.bazel b/stats/grpc_testing/BUILD.bazel
--- a/stats/grpc_testing/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/stats/grpc_testing/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,13 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["test.pb.go"],
+    importpath = "google.golang.org/grpc/stats/grpc_testing",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/status/BUILD.bazel b/status/BUILD.bazel
--- a/status/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/status/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,39 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "go16.go",
+        "go17.go",
+        "status.go",
+    ],
+    importpath = "google.golang.org/grpc/status",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//codes:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@com_github_golang_protobuf//ptypes:go_default_library_gen",
+        "@go_googleapis//google/rpc:status_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "go17_test.go",
+        "status_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = [
+        "//codes:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@com_github_golang_protobuf//ptypes:go_default_library_gen",
+        "@go_googleapis//google/rpc:code_go_proto",
+        "@go_googleapis//google/rpc:errdetails_go_proto",
+        "@go_googleapis//google/rpc:status_go_proto",
+        "@io_bazel_rules_go//proto/wkt:any_go_proto",
+        "@io_bazel_rules_go//proto/wkt:duration_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/stress/client/BUILD.bazel b/stress/client/BUILD.bazel
--- a/stress/client/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/stress/client/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,26 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/stress/client",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//codes:go_default_library",
+        "//credentials:go_default_library",
+        "//grpclog:go_default_library",
+        "//interop:go_default_library",
+        "//interop/grpc_testing:go_default_library",
+        "//status:go_default_library",
+        "//stress/grpc_testing:go_default_library",
+        "//testdata:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "client",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/stress/grpc_testing/BUILD.bazel b/stress/grpc_testing/BUILD.bazel
--- a/stress/grpc_testing/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/stress/grpc_testing/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,13 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["metrics.pb.go"],
+    importpath = "google.golang.org/grpc/stress/grpc_testing",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/stress/metrics_client/BUILD.bazel b/stress/metrics_client/BUILD.bazel
--- a/stress/metrics_client/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/stress/metrics_client/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,20 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "google.golang.org/grpc/stress/metrics_client",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//:go_default_library",
+        "//grpclog:go_default_library",
+        "//stress/grpc_testing:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "metrics_client",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/tap/BUILD.bazel b/tap/BUILD.bazel
--- a/tap/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/tap/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,9 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["tap.go"],
+    importpath = "google.golang.org/grpc/tap",
+    visibility = ["//visibility:public"],
+    deps = ["@org_golang_x_net//context:go_default_library"],
+)
diff -urN a/test/bufconn/BUILD.bazel b/test/bufconn/BUILD.bazel
--- a/test/bufconn/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/test/bufconn/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,14 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["bufconn.go"],
+    importpath = "google.golang.org/grpc/test/bufconn",
+    visibility = ["//visibility:public"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["bufconn_test.go"],
+    embed = [":go_default_library"],
+)
diff -urN a/test/BUILD.bazel b/test/BUILD.bazel
--- a/test/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/test/BUILD.bazel	2018-09-27 16:55:16.114787111 -0400
@@ -0,0 +1,60 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "race.go",
+        "rawConnWrapper.go",
+        "servertester.go",
+    ],
+    importpath = "google.golang.org/grpc/test",
+    visibility = ["//visibility:public"],
+    deps = [
+        "@org_golang_x_net//http2:go_default_library",
+        "@org_golang_x_net//http2/hpack:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "channelz_linux_go110_test.go",
+        "channelz_test.go",
+        "end2end_test.go",
+        "gracefulstop_test.go",
+        "retry_test.go",
+    ],
+    embed = [":go_default_library"],
+    deps = [
+        "//:go_default_library",
+        "//balancer/grpclb:go_default_library",
+        "//balancer/roundrobin:go_default_library",
+        "//codes:go_default_library",
+        "//connectivity:go_default_library",
+        "//credentials:go_default_library",
+        "//encoding/gzip:go_default_library",
+        "//grpclog/glogger:go_default_library",
+        "//health:go_default_library",
+        "//health/grpc_health_v1:go_default_library",
+        "//internal:go_default_library",
+        "//internal/channelz:go_default_library",
+        "//internal/envconfig:go_default_library",
+        "//internal/leakcheck:go_default_library",
+        "//keepalive:go_default_library",
+        "//metadata:go_default_library",
+        "//peer:go_default_library",
+        "//resolver:go_default_library",
+        "//resolver/manual:go_default_library",
+        "//resolver/passthrough:go_default_library",
+        "//stats:go_default_library",
+        "//status:go_default_library",
+        "//tap:go_default_library",
+        "//test/grpc_testing:go_default_library",
+        "//testdata:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@go_googleapis//google/rpc:status_go_proto",
+        "@io_bazel_rules_go//proto/wkt:any_go_proto",
+        "@org_golang_x_net//context:go_default_library",
+        "@org_golang_x_net//http2:go_default_library",
+    ],
+)
diff -urN a/test/codec_perf/BUILD.bazel b/test/codec_perf/BUILD.bazel
--- a/test/codec_perf/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/test/codec_perf/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,9 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["perf.pb.go"],
+    importpath = "google.golang.org/grpc/test/codec_perf",
+    visibility = ["//visibility:public"],
+    deps = ["@com_github_golang_protobuf//proto:go_default_library"],
+)
diff -urN a/test/grpc_testing/BUILD.bazel b/test/grpc_testing/BUILD.bazel
--- a/test/grpc_testing/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/test/grpc_testing/BUILD.bazel	2018-09-27 16:55:16.110787059 -0400
@@ -0,0 +1,13 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["test.pb.go"],
+    importpath = "google.golang.org/grpc/test/grpc_testing",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+        "@org_golang_x_net//context:go_default_library",
+    ],
+)
diff -urN a/testdata/BUILD.bazel b/testdata/BUILD.bazel
--- a/testdata/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/testdata/BUILD.bazel	2018-09-27 16:55:16.114787111 -0400
@@ -0,0 +1,8 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["testdata.go"],
+    importpath = "google.golang.org/grpc/testdata",
+    visibility = ["//visibility:public"],
+)
